### PR TITLE
Enhance Ghost comment mitigation Settings

### PR DESCRIPTION
**Pull request by 6543**
a refactor ... and follow up of  #14349

### CODE REVIEW

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -647,7 +647,7 @@ repos_none = You do not own any repositories
 
 delete_account = Delete Your Account
 delete_prompt = This operation will permanently delete your user account. It <strong>CAN NOT</strong> be undone.
-delete_with_all_comments = Your account is younger than %d days. To avoid ghost comments, all issue/PR comments will be deleted with it.
+delete_with_all_comments = Your account is younger than %d. To avoid ghost comments, all issue/PR comments will be deleted with it.
```
Comment (CirnoT) [inline options/locale/locale_en-US.ini]: Should be `%s` I believe

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -474,7 +474,7 @@ relation to port exhaustion.
 - `ALLOW_ONLY_EXTERNAL_REGISTRATION`: **false** Set to true to force registration only using third-party services.
 - `NO_REPLY_ADDRESS`: **DOMAIN** Default value for the domain part of the user's email address in the git log if he has set KeepEmailPrivate to true.
   The user's email will be replaced with a concatenation of the user name in lower case, "@" and NO_REPLY_ADDRESS.
-- `USER_DELETE_WITH_COMMENTS_MAX_DAYS`: **0** If a user deletes his account before that amount of days, his comments will be deleted as well.
+- `USER_DELETE_WITH_COMMENTS_MAX_TIME`: **2h** If a user deletes his account before that amount of days, his comments will be deleted as well. 0 to disable it.
```
Comment (CirnoT) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]: Really unsure if this is a good idea at all

Comment (6543) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]: @CirnoT do you think it should be disabled by default?

Comment (CirnoT) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]: I do believe this solution is a very specific use-case one and indeed should not be enabled by default.

Proper interaction limits like GH has would be a better solution and would make this one unnecessary in future.

Comment (lafriks) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md]: I agree this should be disabled by default

**Comment by 6543** [comment]
@CirnoT @lafriks Disabled by Default

File: models/admin.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func removeStorageWithNotice(e Engine, bucket storage.ObjectStorage, title, path
 	if err := bucket.Delete(path); err != nil {
 		desc := fmt.Sprintf("%s [%s]: %v", title, path, err)
 		log.Warn(title+" [%s]: %v", path, err)
-		if err = createNotice(x, NoticeRepository, desc); err != nil {
+		if err = createNotice(e, NoticeRepository, desc); err != nil {
```
Comment (zeripath) [inline models/admin.go:78]: This is CRITICAL - it needs to be backported to 1.13

File: custom/conf/app.example.ini [hunk 4]
```diff
@@ -690,7 +690,7 @@ AUTO_WATCH_NEW_REPOS = true
 AUTO_WATCH_ON_CHANGES = false
 ; Default value for the minimum age a user has to exist before deletion to keep issue comments.
 ; If a user deletes his account before that amount of days, his comments will be deleted as well.
-USER_DELETE_WITH_COMMENTS_MAX_DAYS = 0
+USER_DELETE_WITH_COMMENTS_MAX_TIME = 0
```
Comment (zeripath) [inline custom/conf/app.example.ini]: need to update comment with this.

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 5]
```diff
@@ -474,7 +474,7 @@ relation to port exhaustion.
 - `ALLOW_ONLY_EXTERNAL_REGISTRATION`: **false** Set to true to force registration only using third-party services.
 - `NO_REPLY_ADDRESS`: **DOMAIN** Default value for the domain part of the user's email address in the git log if he has set KeepEmailPrivate to true.
   The user's email will be replaced with a concatenation of the user name in lower case, "@" and NO_REPLY_ADDRESS.
-- `USER_DELETE_WITH_COMMENTS_MAX_DAYS`: **0** If a user deletes his account before that amount of days, his comments will be deleted as well.
+- `USER_DELETE_WITH_COMMENTS_MAX_TIME`: **0** If a user deletes his account before that amount of days, his comments will be deleted as well.
```
Comment (zeripath) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md]: Need to update comment with this.

File: models/admin.go (hunk 3 above)
Comment (6543) [reply models/admin.go:78]: yes ... will do backport when this got merged

File: services/comments/comments.go [hunk 6]
```diff
@@ -44,7 +44,7 @@ func UpdateComment(c *models.Comment, doer *models.User, oldContent string) erro
 
 // DeleteComment deletes the comment
 func DeleteComment(comment *models.Comment, doer *models.User) error {
```
Comment (zeripath) [inline services/comments/comments.go]: I think we should switch these arguments around

File: models/issue_comment.go [hunk 7]
```diff
@@ -1037,33 +1037,42 @@ func UpdateComment(c *Comment, doer *User) error {
...
-	if _, err := sess.Delete(&Comment{
+	if err := deleteComment(sess, comment); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func deleteComment(e Engine, comment *Comment) error {
+
```
Comment (zeripath) [inline models/issue_comment.go]:
```suggestion
```

**Review by zeripath — APPROVED** [review]
Otherwise LGTM

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
Please send partial backport :)

**Comment by 6543** [comment]
@techknowlogick Backport -> #14425

**Pull request merged**